VMEncryption/extension_shim.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+
+# Keeping the default command
+COMMAND=""
+PYTHON=""
+
+USAGE="$(basename "$0") [-h] [-i|--install] [-u|--uninstall] [-d|--disable] [-e|--enable] [-p|--update] [-m|--daemon]
+
+Program to find the installed python on the box and invoke a Python extension script using Python 2.7.
+
+where:
+    -h|--help       show this help text
+    -i|--install    install the extension
+    -u|--uninstall  uninstall the extension
+    -d|--disable    disable the extension
+    -e|--enable     enable the extension
+    -p|--update     update the extension
+    -m|--daemon     invoke daemon option
+    -c|--command    command to run
+
+example:
+# Install usage
+$ bash extension_shim.sh -i
+python ./main/handle.py -install
+
+# Custom executable python file
+$ bash extension_shim.sh -c ""hello.py"" -i
+python hello.py -install
+
+# Custom executable python file with arguments
+$ bash extension_shim.sh -c ""hello.py --install""
+python hello.py --install
+"
+
+function find_python(){
```

Comment:
You aren't planning to add Python3 check explicitely?

```
function find_python(){
    local python_exec_command=$1

    # Check if there is python defined.
    if command -v python >/dev/null 2>&1 ; then
        eval ${python_exec_command}="python"
    else
        # Python was not found. Searching for Python3 now.
        if command -v python3 >/dev/null 2>&1 ; then
            eval ${python_exec_command}="python3"
        fi
    fi
}
```

Reply:
The extension code that this shim calls is not yet compatible with Python3.  We are hoping to make extension code Python3 compatible in the future.  Once it is, then we'd add it to the shim too.

Reply:
You can use [2to3](https://docs.python.org/2/library/2to3.html) to do most of your work (transform from 2 to 3) and test it with a test version of the extension (Copy on the box and run in ad-hoc way). I used that for VMAccess for Linux, and HandlerUtils have been changed for the same as well - and then had to do very minor changes. Something for you to consider.